xdsl/irdl/irdl.py

```diff
@@ -1148,6 +1165,17 @@ def get_constraint(
 
                 raise wrong_field_exception(field_name)
 
+        op_def.assembly_format = pyrdl_def.assembly_format
+        if op_def.assembly_format is not None and (
+            pyrdl_def.print != Operation.print
+            or not hasattr(pyrdl_def.parse, "__func__")
+            or pyrdl_def.parse.__func__ != Operation.parse.__func__
```

Comment:
I'm not sure why we are checking for `__func__`?

We can use `getattr(pyrdl_def.parse, "__func__", None) != getattr(Operation.parse, "__func__", None)`

Reply:
What would be the advantage of this change?

Reply:
`pyright` is unhappy about accessing `__func__`

Reply:
This is caught and reported by the CI, no?

Reply:
It is reported by CI so I'm suggesting ways to solve it.

Reply:
Because if one use `hasattr` and then access the attribute, mostly it can combined into `getattr`.

Reply:
That's a good point. I wonder if there's some pyright-blessed way to do this, otherwise might as well do `getattr`